patterns/behavioural/RAII.md

```diff
@@ -108,6 +127,25 @@ Note that implementing `Deref` is not a core part of this pattern, it only makes
 using the guard object more ergonomic. Implementing a `get` method on the guard
 works just as well.
 
+When compared with RAII in C++, there are a few significant differences:
+
+* while C++ code often interfaces with C code or code in older styles, which doesn't use RAII, Rust was designed without the need to interface with such code, so its far less common to implement OBRM yourself
+* C++ doesn't have `Deref` nor a borrow checker, so code using RAII can not archive the same combination of safety and ergonomics
+* perhaps most importantly, Rust has different semantics when it comes to moving and copying of values, this will be expanded on below.
+
+C++ has complex rules for copying and moving of values, that Rust managed to simplify while keeping most advantages.
+In C++ behavior on a "move" (which is semantically meant to signify passing held resources to the moved-to value) is customizable in its move and move-assignment constructors.
+But after a variable has been "moved out of", it must still be accessable in C++.
+In Rust, a moved-out-of variable can not be used, only reassigned a new value (this is referred to as "destructive move"), and the behavior on a move is not customizable, instead a move simply copies the bytes of the moved-out value into the moved-into variable, and ensures the semantics of a destructive move.
+
+<!-- TODO this should be improved, I find it difficult to separate the creation and management of RAII Objects in the **-constructor - so at declaration time - from the one when using the RAII object. Feedbak welcome. -->
+This massively simplifies creation and management of OBRM Objects compared to C++, where one often has to do a lot more manual management of RAII classes - definition of the `destructor`, the `copy constructor`, the `copy assignment constructor`, the `move constructor` and the `move assignment constructor` all at once -, which is very error prone, and where RAII objects have to have a legal moved-out state, which often makes usage of these classes more problematic.
+For example, `unique_ptr`, the C++ equivalent to `Box`, can contain `nullptr`.
```

Comment:
... and thus is more comparable to an `Option<Box>`, but less safe.

Reply:
In terms of how its working under the hood if used correctly, yes. But many people work under the assumtion that unique_ptr will always or at least mostly not contain nullptr.

I'm not certain how to express that without overcomplications, and in the end, unique_ptr was introduced with similar aims to box, so I kept it simple at the expense of being perhapos not entirely correct.

Reply:
We could fix the correctness though, by not saying unique_ptr is equivalent, but instead something like that it was introduced for the same purpose.

Reply:
Yes, that would also be fine.

Reply:
Rewrote it, but perhaps you can think of a more fluent way to express it. Otherwise feel free to resolve this.